Task06_Yoav_Uzan/src/main.cpp

```diff
@@ -0,0 +1,153 @@
+#include <functional>
+#include <vector>
+#include <algorithm>
+#include <cmath>
+#include <iostream>
+#include <map>
+#include <string>
+#include "Location.cpp"
+#include <fstream>
+#include <filesystem>
+#include "SortByMap.cpp"
+
+/**
+  brief Calculate the Euclidean distance between two locations.
+  param loc1 The first location.
+  param loc2 The second location.
+  return The Euclidean distance between the two locations.
+ */
+double L2(const Location& loc1, const Location& loc2)
+{
+    // dis = sqrt((x1 - x2)^2 + (y1 - y2)^2)
+    return std::sqrt(std::pow(loc1._latitude - loc2._latitude, 2) + std::pow(loc1._longitude - loc2._longitude, 2));
+}
+
```

Comment:
your code is very complex, every thing is handled in the main
you should create more classes to handle the logic